### DISCUSSION TITLE
[BP#76] Utiliser les procédures stockées

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#72] [Bonnes pratiques] [answered]
Cette BP me semble totalement ancestrale...ok éventuellement pour la taille des requêtes (même si cela me semble assez négligeable à côté des données transmises en lecture ou en écriture).
Par contre tous les serveurs SQL font de la mise en cache de plans d'exécution, ce qui a exactement l'effet de la precompilation d'une proc stoc. L'important pour que cela fonctionne avec du SQL est d'avoir du SQl paramétré (également essentiel niveau sécurité pour éviter par exemple nombre d'injections SQL).
A voir pour la modifier, voire la supprimer/remplacer à mon avis

**Comment by ACTLEM** [comment]
Je suis déjà tombé sur des cas où la procédure stockée est intéressante car limite fortement les A/R serveur HTTP et BDD. Après, cela aurait pu se faire aussi avec une "méga" requête optimisée mais la maintenabilité laissait à désirer. 
En fait, je comprends la remarque car les cas où il y a un intérêt sont plutôt très à la marge d'après mon expérience. Donc intéressé d'avoir d'autres retours sur cette BP.

**Comment by tbroyer** [comment]
Sur le sujet: https://blog.jooq.org/the-cost-of-jdbc-server-roundtrips/
Mais il ne s'agit pas tant d'utiliser des procédures stockées que de déplacer la logique vers la base de données (ça pourrait être une requête SQL complexe) quand elle traite un grand nombre de données.
Ce n'est pas l'esprit de la BP ici qui parle d'éviter d'envoyer au serveur des requêtes complexe, mais ça pourrait éventuellement la remplacer: éviter le transfert d'une grande quantité de données pour faire un traitement si ce traitement peut être déporté plus près des données. Ça s'applique aussi bien à une base de données qu'un micro-service ou même backend vs frontend (même si souvent dans ce dernier cas ce sera fait, parce que le réseau ici est Internet et qu'on le considère différemment d'un réseau interne).

**Reply by devcrafting** [reply]
J'aime bien l'idée formulée ainsi car beaucoup moins centrée sur les procs stocs et donc plus largement applicable.

**Reply by ACTLEM** [reply]
Du même avis que @devcrafting :-)

**Reply by tbroyer** [reply]
Voir aussi sur ce sujet: https://blog.jooq.org/10-common-mistakes-java-developers-make-when-writing-sql/#:~:text=Processing%20data%20in%20Java%20memory

**Comment by ACTLEM** [comment, marked as answer]
https://github.com/cnumr/best-practices/issues/146
